lib/pbench/client/oidc_admin.py

```diff
@@ -12,7 +12,7 @@ class OIDCAdmin(Connection):
     ADMIN_PASSWORD = os.getenv("OIDC_ADMIN_PASSWORD", "admin")
 
     def __init__(self, server_url: str):
-        super().__init__(server_url, verify=False)
+        super().__init__(server_url, verify=os.environ.get("CURL_CA_BUNDLE", False))
```

Comment:
I _think_ we want the default to be `True`.  That is, if `CURL_CA_BUNDLE` doesn't point at a custom CA, we should be verifying the TLS cert using publicly-available CA's, by default.

I wonder if Requests is clever enough to convert a `"False"` string to a boolean if you pass it into `verify` (e.g., if it cannot open it as a file...).

Reply:
Aha, the problem is that this environment variable `CURL_CA_BUNDLE` that we set in `run-server-func-tests` is not available inside the CI image to use. We have to map this CA cert inside a CI container and then use `env` to set an environment variable for the CI container to use. So everywhere inside the client code when we do this `verify=os.environ.get("CURL_CA_BUNDLE", False)` we are actually bypassing the cert verification (because the value would be `False`).  

But that's not it, setting the `CURL_CA_BUNDLE` env variable on the CI container does not work either because it overrides the default certificate location and then fails to install the test requirements (It could not fetch URL https://pypi.org/simple/bottle/). Therefore we need to use a custom env variable like `OIDC_CA` and then check in the client code whether it exists and if it does then use it.

Reply:
Ugh!  So, yeah, we want the default to be `True`....  🥴 

We need to ponder how we get the CA for the three places we use it:  the Pbench Client, the OIDC client, and the deployment scripts.  I suspect that we'll want different answers in the different places.

Reply:
> setting the `CURL_CA_BUNDLE` env variable on the CI container does not work

Right...that's _way_ too big a scope!  It should be set just on the Pbench Server or possibly just on the Pbench Server's container.

Reply:
> We have to map this CA cert inside a CI container

As I said elsewhere, I really think that it will be better to build the CA into the container rather than mapping it.  The problem is that we want to be able to use a common deployment for development/func-test, Staging, and Production, which means that it would be best if the deployment command/script were the same in all three cases.  This means that the difference has to be in the _contents_ of the things which we map in, not in the number or names of them.  That drives us toward using the Pbench Server config file to dictate how TLS verification is done, rather than using environment variables, and it means that the CA needs to already be in the container for when we want to use it.

Reply:
Getting a CA certificate inside a server container is not a problem we can do that when we build the server image, the problem is getting it inside the CI image where we run the client code. Our client makes a connection request to OIDC broker (Keycloak) for registering a test_user and logging in that's why I was saying we need to map CA cert inside a CI container (since we don't always build it)

> I think Tox is probably getting in the way

Yeah, we need to use `passenv` of the tox.ini if we want to pass certain env variables for the test.

Reply:
> the problem is getting it inside the CI image where we run the client code.

This shouldn't be a problem, because (I think...) we run the client code out of a Git checkout, and the CA is in the Git checkout, so we _should_ just have to reference it (when we want it).

> Yeah, we need to use `passenv` of the `tox.ini`

Yeah 😒, do you feel like adding that?

Reply:
Yeah well that is true we do run the client code out of a Git checkout so we should be able to just reference it.

Reply:
The question is in the details of _how_ we reference it, and I think the best solution for that is the Pbench Server config file.